internal/xds/rbac/rbac_engine.go

```diff
@@ -310,9 +274,17 @@ type rpcData struct {
 }
 
 func (e *engine) doAuditLogging(rpcData *rpcData, rule string, authorized bool) {
+	// In the RBAC world, we need to have a SPIFFE ID as the principal for this
+	// to be meaningful
+	principal := ""
+	if len(rpcData.certs) != 0 && len(rpcData.certs[0].URIs) != 0 &&
```

Comment:
Are we certain `rpcData` is definitely not nil? If not please perform nil check to avoid panic. `certs` being nil is ok because `len(nil)` returns 0.

Reply:
How do we want this to behave? We definitely have RPC data if we get here from the call to `newRPCData`, which will return either `nil, error` or good RPC data.

I think it's good defensive coding anyways to check for `nil` here, but then what do we do? Immediately return?`doAuditLogging` is designed to not return errors, so we just swallow it?

Reply:
This is not a public API so as long as we are sure I personally think it's fine to leave it as is.